dynamics/apis/invoice_line_items.py

```diff
@@ -36,3 +50,43 @@ def delete(self, purchase_invoice_lineitem_id: str, **kwargs):
         :return:
         """
         return self._delete_request({**kwargs}, PurchaseInvoiceLineItems.DELETE_PURCHASE_INVOICE_LINEITEM.format(purchase_invoice_lineitem_id))
+
+    def create_purchase_invoice_line_items(self, purchase_invoice_id: str, line_items: list):
+        """
+        Create PurchaseInvoice LineItems in bulk.
+
+        :param purchase_invoice_id: The ID of the purchase invoice.
+        :param line_items: A list of line items to be added to the purchase invoice.
+        :return: Bulk response containing the results of the bulk post operation.
+        """
+        # Prepare payload for bulk post
+        bulk_payload = []
+        for line_item in line_items:
+            # Prepare payload for each line item
+            line_item_payload = {
+                "method": "POST",
+                "url": PurchaseInvoiceLineItems.BULK_POST_PURCHASE_INVOICE_LINEITEM.format(purchase_invoice_id),
+                "headers": {
+                    "CompanyId": self.__company_id,
+                    "Content-Type": "application/json",
+                    "If-Match": "*"
+                },
+                "body": line_item
+            }
+            bulk_payload.append(line_item_payload)
+
+        # Create a bulk payload containing all line item requests
+        bulk_request_payload = {'requests': bulk_payload}
+
+        # Make the bulk post request
+        bulk_payload_response = self._bulk_post_request(bulk_request_payload)
+
+        # Check the responses for any errors
+        for response in bulk_payload_response['responses']:
```

Comment:
Done add this in this function, this should be handled at the application level, SDK is more general purpose

Reply:
remove any deletion logic from here